mcda/utils/utils_for_main.py

```diff
@@ -426,17 +464,25 @@ def save_df(df: pd.DataFrame, folder_path: str, filename: str):
     :param filename: str
     :return: None
     """
+    custom_output_path = os.environ.get('PROMCDA_OUTPUT_PATH')  # check if an environmental variable is set
+    output_directory_path = custom_output_path if custom_output_path else DEFAULT_OUTPUT_DIRECTORY_PATH
+
     timestamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
     new_filename = f"{timestamp}_{filename}"
 
-    result_path = os.path.join(folder_path, new_filename)
-    df.to_csv(path_or_buf=result_path, index=False)
+    full_output_path = os.path.join(output_directory_path, folder_path, new_filename)
+    ensure_directory_exists(os.path.dirname(full_output_path))
```

Comment:
- What if the user just wrote his name or some junk characters instead of giving you the path? You will get an exception creating dir(s).

Reply:
I added:

    ```
if not os.path.isdir(folder_path):
        logging.error(f"The provided folder path '{folder_path}' is not a valid directory.")
        return
```

Reply:
Same logic was added also in `save_config` and `save_dict`.

Reply:
I don't think it is a good idea to check `isdir` for `folder_path`. This way you are validating if the folder path already exists. This is not what we want. 

I meant just to catch exception from `ensure_directo.....()` if the path provided by the user cannot be used to create directories then report an error.